lib/puppet/provider/package/chocolatey.rb

```diff
@@ -128,7 +128,7 @@ def install
       end
 
       # Add the package version
-      args << @resource[:name][/\A\S*/] << '-version' << @resource[:ensure]
+      args << @resource[:name][/\A\S*/] << '--version' << @resource[:ensure]
```

Comment:
The problem with taking this without version checking Chocolatey is that is makes this provider incompatible with Chocolatey v0.9.8.x and below.